src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,113 @@
+package mate.academy.dao.impl;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.ConnectionUtil;
+import mate.academy.dao.BookDao;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.service.DataProcessingException;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows == 0) {
+                throw new DataProcessingException("Expected to insert at least one row,"
+                        + " but inserted 0 rows");
+            }
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can' add a new book with id: " + book.getId());

```

Comment:
The error message in the `DataProcessingException` should be more informative. Instead of appending the book ID (which is not set when the book hasn't been inserted yet), include the book title and price in the message.